pkg/operator/ceph/cluster/osd/provision_spec.go

```diff
@@ -172,6 +173,7 @@ func (c *Cluster) provisionPodTemplateSpec(osdProps osdProperties, restart v1.Re
 
 func (c *Cluster) provisionOSDContainer(osdProps osdProperties, copyBinariesMount v1.VolumeMount, provisionConfig *provisionConfig) (v1.Container, error) {
 	envVars := c.getConfigEnvVars(osdProps, k8sutil.DataDir)
+	envVars = append(envVars, k8sutil.NodeEnvVar(), mon.PodNamespaceEnvVar(c.clusterInfo.Namespace))
```

Comment:
Why not removing the duplication of envVars after appending all variables? Then it's not necessary to bypass [this k8s issue(https://github.com/kubernetes/kubernetes/issues/86163) and can detect "failed to calculate diff" that doesn't come from this issue. Although this removal is O(n^2), n is not so big and this function is called not so often.

Reply:
Good suggestion! Tracking down the duplicate env vars is actually surprisingly complex, so it would simplify the code if we just always automatically removed the duplicates for OSDs. Other daemons aren't so complex. We can actually implement as O(n) by using a map, similar to how the unit test was implemented.